src/App.tsx

```diff
@@ -1,8 +1,115 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import {
+  getComments, getPosts, getUsers, Posts,
+} from './helpers/api';
+import { LoadingSpinner } from './loading-spinner';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const App = () => {
+  const [post, setPost] = useState<Posts[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [sortedPosts, setSortedPosts] = useState<Posts[]>([]);
+
+  const handleLoad = async () => {
+    setIsLoading(true);
+    try {
+      const postsFromServer = await getPosts();
+      const usersFromServer = await getUsers();
+      const commentsFromServer = await getComments();
+
+      const allPosts = postsFromServer.map(newPost => ({
+        ...newPost,
+        user: usersFromServer.find(user => user.id === newPost.userId),
+        comments: commentsFromServer.filter(comment => comment.postId === newPost.id),
+      }));
+
+      setPost(allPosts);
+      setIsLoaded(true);
+      setSortedPosts(allPosts);
+    } catch (e) {
+      setErrorMessage('Loading from server terminated. Please try again later');
+    }
+
+    setIsLoading(false);
+  };
+
+  const sortPosts = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSortedPosts(post.filter(el => ((el.title + el.body)
+      .toLocaleLowerCase()
+      .includes((event.target.value)
+        .toLocaleLowerCase()))));
+  };
+
+  return (
+    <div className="main">
+      <h1 className="main__header">Dynamic list of posts</h1>
+      {!isLoaded ? (
+        <>
+          {!isLoading
+            ? (
+              <button className="main__load-button" disabled={isLoading} type="button" onClick={handleLoad}>
```

Comment:
keep the line no more than 80 characters;

if you have more than 2 attributes, place each one on the new line

Reply:
fixed